src/storage/disk_table.cc

```diff
@@ -1097,8 +1158,138 @@ void DiskTableRowIterator::SeekToFirst() {
+
+        index_def = std::make_shared<IndexDef>(column_key.index_name(), table_index_.GetMaxIndexId() + 1,
+                IndexStatus::kReady, ::openmldb::type::IndexType::kTimeSerise, col_vec);
+
+
+        if (!column_key.ts_name().empty()) {
+            auto ts_iter = schema.find(column_key.ts_name());
+            if (ts_iter == schema.end()) {
+                PDLOG(WARNING, "not found col_name[%s]. tid %u pid %u", column_key.ts_name(), id_, pid_);
+                return false;
+            }
+            index_def->SetTsColumn(std::make_shared<ColumnDef>(ts_iter->second));
+        } else {
+            index_def->SetTsColumn(std::make_shared<ColumnDef>(DEFUALT_TS_COL_NAME, DEFUALT_TS_COL_ID,
+                        ::openmldb::type::kTimestamp, true));
+        }
+
+        if (column_key.has_ttl()) {
+            index_def->SetTTL(::openmldb::storage::TTLSt(column_key.ttl()));
+        } else {
+            index_def->SetTTL(*(table_index_.GetIndex(0)->GetTTL()));
+        }
+
+        if (table_index_.AddIndex(index_def) < 0) {
+            PDLOG(WARNING, "add index failed. tid %u pid %u", id_, pid_);
+            return false;
+        }
+
+        index_def->SetInnerPos(inner_id);
+        std::vector<std::shared_ptr<IndexDef>> index_vec = {index_def};
+        auto inner_index_st = std::make_shared<InnerIndexSt>(inner_id, index_vec);
+        table_index_.AddInnerIndex(inner_index_st);
+        table_index_.SetInnerIndexPos(new_table_meta->column_key_size() - 1, inner_id);
+
+        rocksdb::ColumnFamilyOptions cfo;
+
+        if (storage_mode_ == ::openmldb::common::StorageMode::kSSD) {
+            cfo = rocksdb::ColumnFamilyOptions(ssd_option_template);
+        } else {
+            cfo = rocksdb::ColumnFamilyOptions(hdd_option_template);
+        }
+
+        cfo.comparator = &cmp_;
+        cfo.prefix_extractor.reset(new KeyTsPrefixTransform());
+
+        if (index_def->GetTTLType() == ::openmldb::storage::TTLType::kAbsoluteTime ||
+            index_def->GetTTLType() == ::openmldb::storage::TTLType::kAbsOrLat) {
+            cfo.compaction_filter_factory = std::make_shared<AbsoluteTTLFilterFactory>(inner_index_st);
+        }
+
+        rocksdb::ColumnFamilyHandle* handle;
+        cf_ds_.push_back(rocksdb::ColumnFamilyDescriptor(index_def->GetName(), cfo));
+        DEBUGLOG("add cf_name %s. tid %u pid %u", index_def->GetName().c_str(), id_, pid_);
+
+        rocksdb::Status s = db_->CreateColumnFamily(cfo, index_def->GetName(), &handle);
+        if (!s.ok()) {
+            PDLOG(WARNING, "rocksdb failed to create ColumnFamily. tid %u pid %u error %s",
+                  id_, pid_, s.ToString().c_str());
+            return false;
+        }
+
+        // may change cf_hs_->data
+        {
+            std::lock_guard<std::mutex> guard(cf_hs_mutex);
+            auto new_cf_hs_ = std::make_shared<std::vector<rocksdb::ColumnFamilyHandle*>>(GetCF_DS_());
```

Comment:
if use `mutex lock`, then no need to copy and reset atomically.

Reply:
As I comment here, `append` may trigger reallocation and modify `cf_hs_->data`. It may cause errors when functions like `NewWindowIterator` and `NewTraverseIterator` accessing elements of `cf_hs_`  by 'old' `cf_hs_->data`. So atomical copy and reset are needed.

Reply:
oh. Yes. You're right. 

In this case, you may need to atomically load the `cf_hs_` in other `Reading` cases.

Reply:
> Yes, after `atomic_store_explicit`, previous `cf_hs_` is destoried.

Yes, after atomic_store_explicit, previous cf_hs_ is destroyed.
Thanks for reminding.

Reply:
> oh. Yes. You're right.
> 
> In this case, you may need to atomically load the `cf_hs_` in other `Reading` cases.

Yes, after `atomic_store_explicit` is done, previous `cf_hs_` is destroyed.
Thanks for reminding.